### PR TITLE
Add SAML IdP support to license/modules.

### PR DESCRIPTION
**Pull request by mdwn**
The SAML IdP feature has been added to the license and to modules.

### CODE REVIEW

File: api/types/license.go [hunk 1]
```diff
@@ -358,6 +376,8 @@ type LicenseSpecV3 struct {
 	SupportsDatabaseAccess Bool `json:"db,omitempty"`
 	// SupportsDesktopAccess turns desktop access on or off
 	SupportsDesktopAccess Bool `json:"desktop,omitempty"`
+	// SupportsSAMLIdP turns SAMLIdP access on or off
+	SupportsSAMLIdP Bool `json:"saml_idp,omitempty"` //nolint:revive // Because we want this to be IdP.
```
Comment (strideynet) [inline api/types/license.go:380]: I wonder if we can set a global ignore/allow for `IdP` ?

File: api/proto/teleport/legacy/client/proto/authservice.proto [hunk 2]
```diff
@@ -548,6 +548,8 @@ message Features {
   reserved 13; // bool ResourceAccessRequests
   // RecoveryCodes enables recovery codes
   bool RecoveryCodes = 14 [(gogoproto.jsontag) = "recovery_codes"];
+  // SAMLIdP enables the the SAML IdP.
```
Comment (strideynet) [inline api/proto/teleport/legacy/client/proto/authservice.proto:551]:
```suggestion
  // SAMLIdP enables the SAML IdP.
```

**Review by strideynet — APPROVED** [review]
Just one spelling error, looks good otherwise.

File: api/types/license.go (hunk 1 above)
Comment (mdwn) [reply api/types/license.go:380]: I looked into it. Basically the `revive` linter seems not to allow mixed case even if you try to relax the restrictions AFAICT.

**Review by tigrato — APPROVED** [review]
Just run `make grpc` again to update the auto-generated proto files with your latest change

**Comment by mdwn** [comment]
So it looks like we don't need to do this for this feature -- sorry for the noise, all.

**Pull request closed without merging**